File: webapp-java-spring/src/main/java/de/ehmke/repository/PersonRepository.java
package de.ehmke.repository;

import de.ehmke.entity.Person;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Component;

import java.util.List;

/**
 *
 */
@Component
public interface PersonRepository extends CrudRepository<Person, Long>
{

    @Query("SELECT DISTINCT p FROM Person p LEFT OUTER JOIN FETCH p.calendars c LEFT OUTER JOIN FETCH c.appointments")
    List<Person> findAllWithAppointments();

}

File: webapp-php-symfony/src/AppBundle/Repository/PersonRepository.php
<?php

namespace AppBundle\Repository;

use Doctrine\ORM\EntityRepository;

/**
 * Class PersonRepository
 */
class PersonRepository extends EntityRepository
{
    /**
     * @return array
     */
    public function getPersonsWithAppointments()
    {
        return $this->_em->createQuery('SELECT p, b, e FROM AppBundle\Entity\Person p JOIN p.calendars b JOIN b.appointments e')
            ->getResult();
    }
}